src/hitchhiker/tree/utils/gc.cljc

```diff
@@ -0,0 +1,23 @@
+(ns hitchhiker.tree.utils.gc
+  (:require [hitchhiker.tree :as hh]
+            [hitchhiker.tree.node :as n]
+            [hitchhiker.tree.bootstrap.konserve :as kons]
+            [hitchhiker.tree.utils.async :as ha]))
+
+(defn mark
+  "Return a set of all addresses reachable from gc-roots, corresponding to the
+  mark phase of a garbage collector."
+  ([gc-roots]
+   (ha/go-try
+       (loop [to-visit gc-roots
```

Comment:
nit: I usually prefer to destructure at this level instead of doing a (if (seq x ...))) later. 

```clj
(loop [[node & r] gc-roots 
       visited #{}] 
   (if node 
     (do ..... (recur (into r new-nodes) ...))
     visited))
```